### DISCUSSION TITLE
Mount SMB Network Drive using Cli.Wrap returns mount error(13): Permission denied.

### DISCUSSION THREAD

**Discussion by tadessefikadu** [Tyrrrz/CliWrap#159] [Q&A]
Mount SMB Network Drive using Cli.Wrap returns mount error(13): Permission denied.

Command:
mount -t cifs -o username=user@fqdn,password='password',sec=ntlmssp //smbshared.fqdn/foldername  /mountfolder  

I am using CliWrap as below 

```c#
var mountResponse = await Cli.Wrap("mount")
                                                 .WithArguments(new []{
                                                   "-t", 
                                                   "cifs",
                                                   "-o", 
                                                   $"username={username},password="password",sec=ntlmssp",
                                                   "//smbshared.fqdn/foldername",
                                                   "/mountfolder  "
                                                  }) .ExecuteBufferedAsync();
 
```
I get "mount error(13): Permission denied." when this is run using Cli.Wrap, however, when in the same container cmd. It runs with no issue and the mount is created. 

Any suggestion on what I need to look into?

**Comment by Tyrrrz** [comment]
Do you run cmd as admin? Otherwise you can try using `SetCredentials(...)` maybe. Or just wrap `cmd` and call `mount` through `cmd`.

**Reply by tadessefikadu** [reply]
Thanks for the quick reply @Tyrrrz.  It is a linux container and for now the container is running under root user. When I run the command in cmd inside the container, the whoami user is root, it must be running it under the admin user.  I tried this with SetCredentials, passed username 'root', with no password. The error was the same.  Can you elaborate what you mean by 'just wrap `cmd` and call `mount` through `cmd`'?   Is there anything I need to change in the way I am trying to execute the command?

**Reply by Tyrrrz** [reply]
Ah, I see, I misunderstood originally what you meant by "cmd". I'm not sure exactly how, but you need to get your C# code to be executed by root as well. If you try to spawn a process with higher privileges than the parent, then you can't redirect stdout/stderr, so you can't do that in CliWrap. Try to execute the command exactly the same way you're doing it manually, maybe by wrapping a shell?

**Reply by tadessefikadu** [reply]
I have not seen an example of the shell, I can explore that. How do I figure the what user and privileges the parent is running under? Or in general how do we know what context the Cli.Wrap command is running under?

**Reply by Tyrrrz** [reply]
> How do I figure the what user and privileges the parent is running under? Or in general how do we know what context the Cli.Wrap command is running under?

Probably with something like this: https://docs.microsoft.com/en-us/dotnet/api/system.environment.username?view=net-6.0

Sorry, I'm not well versed in Linux stuff
